qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -586,6 +586,8 @@ def start_workbench(app, command_line_options):
             main_window.close()
             return 0
 
+    # Before checking for recovery checkpoints we need to make sure the recovery files are "correct" and not bugged
```

Comment:
Can I ask for some clarification on this? I think if we say "buggy"/"correct" then it sounds like our code is wrong. In fact we have accounted for the situation of having an incomplete checkpoint by just removing and ignoring it. Maybe `repair_checkpoints` should be `remove_locked_checkpoints`?